csharp/src/Client/AdbcCommand.cs

```diff
@@ -114,10 +115,31 @@ public override CommandType CommandType
             }
         }
 
+        /// <summary>
+        /// Gets or setts the name of the command timeout property for the underlying ADBC driver.
```

Comment:
nit: `sets`

That said, I feel like this is a dubious design choice. If there were a standard for command timeout properties in ADBC then we could hook into that from the wrapper. But if the consumer needs to have custom code anyway in order to set the command timeout, then I think we're better off just exposing the ability to set statement options through DbCommand -- it's equally usable for this case, and more generally useful for other unanticipated cases.

Reply:
The challenge with that is that each driver has a different property that needs to be set. Just setting `CommandTime = 30;` won't actually _do_ anything other than cause confusion.

Reply:
Removed `CommandTimeoutProperty`. The consumer can already access `AdbcCommand.AdbcStatement.SetOption(string, string)`. Changed override for `CommandTimeout` to throw `NotImplementedException` - is this a breaking change?.

Reply:
Also note that a particular driver may define this in a unit other than seconds given the lack of a standard.

As a suggestion, if the property name were on the ADO.NET connection string, it could be used much more transparently because the consumer will already have to have a generic mechanism for setting connection string properties. With this API, the consumer has to know explicitly that they're using a wrapped ADBC driver in order to be able to use the `CommandTimeout` API.